nav2_util/include/nav2_util/simple_action_server.hpp

```diff
@@ -36,12 +35,23 @@ class SimpleActionServer
   explicit SimpleActionServer(
     rclcpp::Node::SharedPtr node,
     const std::string & action_name,
-    ExecuteCallback execute_callback)
+    ExecuteCallback execute_callback,
+    bool autostart = true)
   : node_(node), execute_callback_(execute_callback)
   {
+    if (autostart) {
+      server_active_ = true;
+    }
+
     auto handle_goal =
-      [](const rclcpp_action::GoalUUID &, std::shared_ptr<const typename ActionT::Goal>)
+      [this](const rclcpp_action::GoalUUID &, std::shared_ptr<const typename ActionT::Goal>)
       {
+        std::lock_guard<std::mutex> lock(activation_mutex_);
```

Comment:
Global comment: I recommend using std::atomic\<bool\> instead of introducing the mutexes. More concise code and typically better performance, depending on the platform: https://www.arangodb.com/2015/02/comparing-atomic-mutex-rwlocks/

Reply:
In this case, I think the mutex is the right way to go. I think bad things could happen if we don't get all the way through the deactivate sequence before someone activates us again.

Reply:
@crdelsey Fortunately, disallowing reentrancy is already handled by ROS2 in the base class (see ServerBase::execute_goal_request_received()). It would be redundant to introduce another mutex.

Reply:
@crdelsey  There is a mutex in the base class that serializes the callbacks, so one won't be activated again (see ServerBase::execute_goal_request_received, for example). However,, the callback to be concerned about is handle_accepted. This could be called while our own execute thread is running. The data that I see in this class that could be simultaneously accessed by the execute thread and the handle_accepted callback includes: current_handle_,  new_handle_,  and preempt_requested_. So, since there are three variables I'm fine with a mutex in that callback and any functions that are provided to the derived class that access these.

Reply:
> There is a mutex in the base class that serializes the callbacks

Oh yeah. I forgot about that. Still, since this class already had an additional mutex, I think we should reuse it if possible. Having multiple sync points gets so hard to reason about.

Reply:
Will change to use a single mutex. The only consequence of this is that `handle_goal` and `handle_accepted` would not run simultaneously. Which I think is fine for a simple action server.